File: pset6/readability/readability.py
from cs50 import get_string


# Prompts user for text to assess
text = get_string("Text: ")


def main():

    # Calls functions to assign values
    nlets = count_letters()
    nwords = count_words()
    nsents = count_sents()

    # Coleman-Liau index formula
    index = round(100 * (0.0588 * nlets / nwords - 0.296 * nsents / nwords) - 15.8)

    # Prints grade based on index
    if index < 1:
        print("Before Grade 1")
    elif index < 16:
        print(f"Grade {index}")
    else:
        print("Grade 16+")


# Function to count the number of letters
def count_letters():
    nlets = 0
    for char in text:
        if(char.isalpha()):
            nlets += 1
    return nlets


# Function to count the number of words
def count_words():
    nwords = len(text.split())
    return nwords


# Function to count the number of sentences
def count_sents():
    nsents = 0
    for char in text:
        if(char == '.' or char == '!' or char == '?'):
            nsents += 1
    return nsents


main()

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int count_letters(void);
int count_words(void);
int count_sent(void);

int nlet, nword, nsent, i, index;
string text;

int main(void)
{
    //Prompts user for text
    text = get_string("Text: ");
    printf("%s\n", text);

    //calls function to return the number of letter, words and sentences
    count_letters();
    count_words();
    count_sent();

    //Prints the readability of the text using the Coleman-Liau index
    index = round(100 * (0.0588 * nlet / (float) nword - 0.296 * nsent / (float) nword) - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index < 16)
    {
        printf("Grade %i\n", index);
    }
    else
    {
        printf("Grade 16+\n");
    }

}


//Function returns the number of letters of the alphabet in a string
int count_letters(void)
{
    for (i = 0, nlet = 0; text[i] != '\0'; i++)
    {
        if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122))
        {
            nlet++;
        }
    }
    return nlet;
}

//Function returns the number of words in a string
int count_words(void)
{
    if ((text[0] >= 65 && text[0] <= 90) || (text[0] >= 97 && text[0] <= 122))
    {
        for (i = 0, nword = 1; text[i] != '\0'; i++)
        {
            if (text[i] == 32)
            {
                nword++;
            }
        }
    }
    else
    {
        nword = 0;
    }

    return nword;

}

//Function returns the number of sentences
int count_sent(void)
{
    for (i = 0, nsent = 0; text[i] != '\0'; i++)
    {
        if (text[i] == 46 || text[i] == 33 || text[i] == 63)
        {
            nsent++;
        }
    }
    return nsent;
}